### PR TITLE
Fix segmentation faults when calling artifact methods

### PR DESCRIPTION
**Pull request by alex-schneider**
- @::getImageArtifact()@
- @::deleteImageArtifact()@

Env: PHP 7.1.2 + Ubuntu

### CODE REVIEW

**Comment by Danack** [comment]
Thanks, the patch looks good. I was going to say that the test probably wouldn't work on PHP 5.3, but it appears to be passing, which is odd.

*makes mental note to check all the other methods that accept string params.*

File: tests/276_Imagick_artifacts.phpt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+--TEST--
+Test artifacts
+--SKIPIF--
+<?php
+
+require_once(dirname(__FILE__) . '/skipif.inc');
+checkClassMethods('Imagick', array('setImageArtifact', 'getImageArtifact', 'deleteImageArtifact'));
+
+/* See imagick_class.c */
+if (Imagick::getVersion()['versionNumber'] < 0x656) {
```
Comment (Danack) [inline tests/276_Imagick_artifacts.phpt]: That line isn't quite valid syntax on PHP 5.3 - https://3v4l.org/j57pb

but the test passes for most of the PHP 5.3 versions. Which is interesting.

Comment (rcanavan) [inline tests/276_Imagick_artifacts.phpt]: So, why does it work with the foo()['..'] syntax over here? And why doesn't the checkClassMethods() check work, i.e. why is there an error that the method does not exist despite the check?

**Comment by Danack** [comment]
I think it looks like that if the skipif section contains invalid syntax then PHP just runs the test.

https://github.com/Danack/imagick/blob/test_php_test/tests/001_imagick_readimage.phpt#L9
vs
https://travis-ci.org/Danack/imagick/jobs/210529659#L4959

which is 'awesome'.

**Pull request merged**